### PR TITLE
ALPHA7

### LINKED ISSUE #1: Import database of other auth plugins

### LINKED ISSUE #2: Add language files support

### PR DESCRIPTION
**Pull request by xXSirGamesXx**
## Introduction

ALPHA7
### Relevant issues

ALPHA7

### CODE REVIEW

File: src/HereAuth/Command/HereAuthCommand.php [hunk 1]
```diff
@@ -38,7 +39,10 @@ public function __construct(HereAuth $main, $name, $desc, $usage, ...$aliases){
 		parent::__construct($name, $desc, $usage, $aliases);
 	}
 
-	public function getMain():HereAuth{
+	/**
+	 * @return HereAuth
+	 */
+	public function getMain(){
```
Comment (robske110) [inline src/HereAuth/Command/HereAuthCommand.php:45]: wtf??

File: src/HereAuth/Command/OptCommand.php [hunk 2]
```diff
@@ -144,7 +144,7 @@ private function getHelpMessage(User $user){
 		$output = "Your HereAuth options:\n";
 		$mlp = $opts->maskLocPos;
 		if(!preg_match(/** @lang RegExp */
-			'#^((\?spawn\?)|((-)?[0-9]+,(-)?[0-9]+,(-)?[0-9]+))@([^/\\\\]+)$#', $mlp, $match)
+			'#^((\?spawn\?)|((\-)?[0-9]+,(\-)?[0-9]+,(\-)?[0-9]+))@([^/\\\\]+)$#', $mlp, $match)
```
Comment (robske110) [inline src/HereAuth/Command/OptCommand.php:147]: why did you change the regex?

File: src/HereAuth/Command/RenameCommand.php [hunk 3]
```diff
@@ -47,7 +47,7 @@ public function run(array $args, CommandSender $sender){
 //			$this->getMain()->getDataBase()->saveData($info, false);
 //			return "Account $args[0] is changed to $args[1]";
 //		}
-		$this->getMain()->getDataBase()->renameAccount($args[0], $args[1], function($result) use ($args, $sender){
+		$this->getMain()->getDataBase()->renameAccount($args[0], $args[1], function ($result) use ($args, $sender){
```
Comment (robske110) [inline src/HereAuth/Command/RenameCommand.php:50]: this is also unnecessary

File: src/HereAuth/Command/UnregisterCommand.php [hunk 4]
```diff
@@ -36,11 +36,11 @@ protected function run(array $args, CommandSender $sender){
 			return $ev->getCancelMessage();
 		}
 		if($user !== null){
-			$user->resetAccount(function($success) use ($sender, $name){
+			$user->resetAccount(function ($success) use ($sender, $name){
```
Comment (robske110) [inline src/HereAuth/Command/UnregisterCommand.php:39]: did you let an IDE do this

File: src/HereAuth/Command/UnregisterCommand.php [hunk 5]
```diff
@@ -36,11 +36,11 @@ protected function run(array $args, CommandSender $sender){
...
 				$sender->sendMessage($success ? "Account $name has been unregistered." : "Account $name does not exist.");
 			});
 		}else{
-			$this->getMain()->getDataBase()->unregisterAccount($name, function($success) use ($sender, $name){
+			$this->getMain()->getDataBase()->unregisterAccount($name, function ($success) use ($sender, $name){
```
Comment (robske110) [inline src/HereAuth/Command/UnregisterCommand.php:43]: here too

File: src/HereAuth/Event/HereAuthEvent.php [hunk 6]
```diff
@@ -23,8 +23,10 @@ public function __construct(HereAuth $main){
 		parent::__construct($main);
 	}
 
-	public function getMain() : HereAuth{
-		/** @noinspection PhpIncompatibleReturnTypeInspection */
+	/**
+	 * @return HereAuth
+	 */
+	public function getMain(){
```
Comment (robske110) [inline src/HereAuth/Event/HereAuthEvent.php:29]: don't remove the typehints

**Review by robske110 — CHANGES_REQUESTED** [review]
You changed some stuff you shouldn't change,
also i don't think this is all that is required, did you test if the plugin works?

File: src/HereAuth/Command/OptCommand.php (hunk 2 above)
Comment (SOF3) [reply src/HereAuth/Command/OptCommand.php:147]: It is actually the same, just more explicit, but unnecessary.

File: src/HereAuth/Command/HereAuthCommand.php [hunk 7]
```diff
@@ -49,11 +53,11 @@ public function getMessage(string $key, string $default) : string{
 	/**
 	 * @return HereAuth
 	 */
-	public function getPlugin(){
+	public function getPlugin() : Plugin{
 		return $this->main;
 	}
 
-	public final function execute(CommandSender $sender, $commandLabel, array $args){
```
Comment (SOF3) [inline src/HereAuth/Command/HereAuthCommand.php:56]: Why remove the final modifier?

File: src/HereAuth/Command/RenameCommand.php (hunk 3 above)
Comment (SOF3) [reply src/HereAuth/Command/RenameCommand.php:50]: Please get rid of IDE work.

File: src/HereAuth/Command/HereAuthCommand.php (hunk 1 above)
Comment (SOF3) [reply src/HereAuth/Command/HereAuthCommand.php:45]: Why did you remove the type hint?

File: src/HereAuth/Event/HereAuthEvent.php (hunk 6 above)
Comment (SOF3) [reply src/HereAuth/Event/HereAuthEvent.php:29]: This change is entirely unnecessary.

**Review by SOF3 — CHANGES_REQUESTED** [review]

File: src/HereAuth/Command/HereAuthCommand.php (hunk 1 above)
Comment (xXSirGamesXx) [reply src/HereAuth/Command/HereAuthCommand.php:45]: Sorry i fixed this a little bit ago, but the branch i was using was the pw-hash branch so i tried pushing it onto this, ima restart
